pkg/oci/containerize.go

```diff
@@ -150,27 +152,32 @@ func newDataTarball(source, target string, ignored []string, verbose bool) error
 			}
 		}
 
-		header, err := tar.FileInfoHeader(info, info.Name())
-		if err != nil {
-			return err
+		var link string
+		if info.Mode()&os.ModeSymlink == os.ModeSymlink {
+			if link, err = os.Readlink(path); err != nil {
+				return err
+			}
+			ok, err := isWithin(source, link)
+			if err != nil {
+				return errors.Wrap(err, "error checking if link is outside function root")
+			} else if !ok {
+				return fmt.Errorf("links outside Function root disallowed.  Link %v references %v", path, link)
+			}
 		}
 
-		relPath, err := filepath.Rel(source, path)
+		header, err := tar.FileInfoHeader(info, link)
 		if err != nil {
 			return err
 		}
 
-		header.Name = slashpath.Join("/func", filepath.ToSlash(relPath))
-		// TODO: should we set file timestamps to the build start time of cfg.t?
-		// header.ModTime = timestampArgument
-
-		if err := tw.WriteHeader(header); err != nil {
+		header.Name = filepath.Join(baseDir, strings.TrimPrefix(path, source))
```

Comment:
Why you use `filepath` instead of `slashpath` here?

Reply:
And why you use `strings.TrimPrefix` instead of `filepath.Rel`?

Reply:
And why is not path slashified by `filepath.ToSlash`?

Reply:
https://github.com/knative/func/actions/runs/7886751749/job/21524127711?pr=2156 -- this is not a flake.

Reply:
Good thing we have the Windows test, sometimes forget the nuances of being Windows-friendy.  Yes, line 173 should have remained the same.